homeassistant/components/zha/light.py

```diff
@@ -201,10 +204,14 @@ def async_restore_last_state(self, last_state):
     async def async_turn_on(self, **kwargs):
         """Turn the entity on."""
         transition = kwargs.get(light.ATTR_TRANSITION)
-        duration = transition * 10 if transition is not None else DEFAULT_DURATION
+        duration = transition * 10 if transition else 0
         brightness = kwargs.get(light.ATTR_BRIGHTNESS)
         effect = kwargs.get(light.ATTR_EFFECT)
 
+        if brightness is None and self._off_brightness is not None:
+            brightness = self._off_brightness
```

Comment:
Should we add `self.brihtness < 15` here as well? I just want if the light was operated from the switch and not from HA and it reported the brightness value, then it should be used instead. This is to ensure the off brightness is used only if it was an off transition.

Reply:
In that case, isn't it better to clear the off_brightness when the lamp reports that it was turned on?

Reply:
Hrm, we could do it in "set_level" if the normalized value > 15 then clear `off_brighness`

Reply:
I agree we should do it there. But I don't think we should check if > 15. If the light goes on, it is on and off_brightness should be cleared.

Reply:
Actually, I think off_brightness should be cleared in async_set_state, if state is True.

Reply:
Good point. We should do both. If the light was actioned externally and we get on_off attribute update -> off_brightness is cleared. If HA successfully called all of the `light.turn_on` service -> clear off_brightness.

The problem with clearing off_brightness only on `set_state()` some lights (HUE) do not report states, so it never would be cleared.

Reply:
and I agree, if we clear `off_brightness` in `set_state()` then no need for `self.brightness > 15`